flink-connector-aws-dynamodb/src/test/java/org/apache/flink/streaming/connectors/dynamodb/util/AWSDynamoDbUtilTest.java

```diff
@@ -0,0 +1,156 @@
+
+        Assertions.assertThat(backoffStrategy).isInstanceOf(EqualJitterBackoffStrategy.class);
+        Assertions.assertThat(
+                        ((EqualJitterBackoffStrategy) backoffStrategy).toBuilder().baseDelay())
+                .isEqualTo(Duration.ofMillis(10));
+        Assertions.assertThat(
+                        ((EqualJitterBackoffStrategy) backoffStrategy).toBuilder().maxBackoffTime())
+                .isEqualTo(Duration.ofMillis(100));
+    }
+
+    @Test
+    public void testConstructsFixedDelayBackoffStrategy() {
+        Properties properties = new Properties();
+        properties.put(AWSDynamoDbConfigConstants.THROTTLING_BACKOFF_STRATEGY, "FIXED_DELAY");
+        properties.put(AWSDynamoDbConfigConstants.FIXED_DELAY_BACKOFF_MS, "10");
+
+        BackoffStrategy backoffStrategy =
+                AWSDynamoDbUtil.getBackoffStrategy(
+                        properties, AWSDynamoDbConfigConstants.THROTTLING_BACKOFF_STRATEGY);
+
+        Assertions.assertThat(backoffStrategy).isInstanceOf(FixedDelayBackoffStrategy.class);
+        Assertions.assertThat(backoffStrategy)
+                .isEqualTo(FixedDelayBackoffStrategy.create(Duration.ofMillis(10)));
+    }
+
+    @Test
+    public void testConstructsRetryPolicy() {
+        Properties properties = new Properties();
+
+        properties.put(AWSDynamoDbConfigConstants.BACKOFF_STRATEGY, "FIXED_DELAY");
+        properties.put(AWSDynamoDbConfigConstants.THROTTLING_BACKOFF_STRATEGY, "FIXED_DELAY");
+
+        int expectedNumRetries = 100;
+        properties.put(
+                AWSDynamoDbConfigConstants.NUMBER_RETRIES, String.valueOf(expectedNumRetries));
+        int expectedDurationMs = 10;
+        BackoffStrategy expectedStrategy =
+                FixedDelayBackoffStrategy.create(Duration.ofMillis(expectedDurationMs));
+        properties.put(
+                AWSDynamoDbConfigConstants.FIXED_DELAY_BACKOFF_MS,
+                String.valueOf(expectedDurationMs));
+
+        RetryPolicy policy = AWSDynamoDbUtil.getRetryPolicy(properties);
+
+        Assertions.assertThat(policy.toBuilder().numRetries()).isEqualTo(expectedNumRetries);
+        Assertions.assertThat(policy.toBuilder().backoffStrategy()).isEqualTo(expectedStrategy);
+        Assertions.assertThat(policy.toBuilder().throttlingBackoffStrategy())
+                .isEqualTo(expectedStrategy);
+    }
+
+    @Test
+    public void testConstructsOverrideConfigurationWithDefaults() {
+        Properties properties = new Properties();
+
+        ClientOverrideConfiguration configuration =
+                AWSDynamoDbUtil.getOverrideConfiguration(properties);
+
+        Assertions.assertThat(
+                        getAdvancedOption(configuration, SdkAdvancedClientOption.USER_AGENT_PREFIX))
+                .isEqualTo(AWSDynamoDbUtil.getFlinkUserAgentPrefix(properties));
+
+        Assertions.assertThat(configuration.retryPolicy().isPresent()).isTrue();
```

Comment:
nit: For a better message when tests fail
```suggestion
        Assertions.assertThat(configuration.retryPolicy()).isEmpty();
```